### ISSUE TITLE
an empty dir config/my_environment results in an error

### ISSUE THREAD

**Issue by rafcio19** [cloudreach/sceptre#138]
If an empty directory exists in `config/my_environment` ie. `config/dev/empty_dir` sceptre will error and not generate/launch a stack

    template_output = env.stacks[stack].template.body
    AttributeError: 'Environment' object has no attribute 'stacks'

**Comment by theseanything** [comment]
@rafcio19 could you supply more details about the error... whats the full structure of your config folder? Whats the exact sceptre command you ran?

**Comment by rafcio19** [comment]
Example structure:
    .
    ├── config
    │   ├── config.yaml
    │   └── dev
    │       ├── sub-env
    │       └── test-config.yaml
    └── templates
        └── test.py

where the presence of `sub-env` will cause this error:

```
sceptre --debug generate-template dev test
[2017-06-12 19:50:57] - sceptre.environment - Initialising 'dev/sub-env' environment...
[2017-06-12 19:50:57] - sceptre.config - Reading in 'config.yaml' files...
[2017-06-12 19:50:57] - sceptre.config - Config: {'region': 'eu-west-1', 'dependencies': []}
Traceback (most recent call last):
  File "/usr/local/bin/sceptre", line 9, in <module>
    load_entry_point('sceptre==1.1.2', 'console_scripts', 'sceptre')()
  File "/Library/Python/2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Library/Python/2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Python/2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Python/2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Library/Python/2.7/site-packages/sceptre/cli.py", line 93, in decorated
    return func(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/sceptre/cli.py", line 167, in generate_template
    template_output = env.stacks[stack].template.body
AttributeError: 'Environment' object has no attribute 'stacks'
```

I'm not sure what the desired behaviour is meant to be here but it stems from line 409 in [environment.py](https://github.com/cloudreach/sceptre/blob/master/sceptre/environment.py#L409)

**Comment by theseanything** [comment]
Fixed in #190
